pkg/util/util.go

```diff
@@ -239,3 +244,147 @@ func SplitQualifiedName(str string) (string, string) {
 func JoinQualifiedName(namespace, name string) string {
 	return path.Join(namespace, name)
 }
+
+type Route struct {
+	Interface   string
+	Destination net.IP
+	Gateway     net.IP
+	// TODO: add more fields here if needed
+}
+
+func getRoutes(input string) ([]Route, error) {
+	routes := []Route{}
+	inFile, err := os.Open(input)
+	if err != nil {
+		return nil, err
+	}
+	defer inFile.Close()
+	scanner := bufio.NewReader(inFile)
+	for {
+		line, err := scanner.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+		if strings.HasPrefix(line, "Iface") {
+			continue
+		}
+		fields := strings.Fields(line)
+		if len(fields) < 11 {
+			return nil, fmt.Errorf("Incorrect number of fields (expected at least 11, got %d): %s", len(fields), line)
+		}
+		routes = append(routes, Route{})
+		route := &routes[len(routes)-1]
+		route.Interface = fields[0]
+		ip, err := parseIP(fields[1])
+		if err != nil {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[1])
+		}
+		route.Destination = ip
+		ip, err = parseIP(fields[2])
+		if err != nil {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[2])
+		}
+		route.Gateway = ip
+	}
+	return routes, nil
+}
+
+func parseIP(str string) (net.IP, error) {
+	bytes, err := hex.DecodeString(str)
+	if err != nil {
+		return nil, err
+	}
+	//TODO add ipv6 support
+	if len(bytes) != net.IPv4len {
+		return nil, fmt.Errorf("only IPv4 is supported")
+	}
+	bytes[0], bytes[1], bytes[2], bytes[3] = bytes[3], bytes[2], bytes[1], bytes[0]
+	return net.IP(bytes), nil
+}
+
+func isInterfaceUp(intf *net.Interface) bool {
+	if intf.Flags&net.FlagUp != 0 {
+		glog.V(4).Infof("Interface %v is up\n", intf.Name)
+		return true
+	}
+	return false
+}
+
+func getFinalIP(addrs []net.Addr) (net.IP, error) {
+	if len(addrs) > 0 {
+		for i := range addrs {
+			glog.V(4).Infof("Checking addr  %s.\n", addrs[i].String())
+			ip, _, err := net.ParseCIDR(addrs[i].String())
+			if err != nil {
+				return nil, err
+			}
+			//Only IPv4
+			//TODO : add IPv6 support
+			if ip.To4() != nil {
+				if !ip.IsLoopback() {
+					glog.V(4).Infof("IP found %v\n", ip)
+					return ip, nil
+				} else {
+					glog.V(4).Infof("Loopback found %v\n", ip)
+				}
+			} else {
+				glog.V(4).Infof("%v is not a valid IPv4 address\n", ip)
+			}
+
+		}
+	}
+	return nil, nil
+}
+
+func getIPFromInterface(intfName string) (net.IP, bool, error) {
```

Comment:
In most of these functions, it seems like you don't need the bool.  If the first return in nil, there was no IP found.  Can you simplify to that all around?